pkg/connector/client.go

```diff
@@ -133,6 +134,10 @@ func (tc *TwitterClient) Connect(ctx context.Context) {
 		}
 	}
 
+	tc.DoConnect(ctx, inboxState)
+}
+
+func (tc *TwitterClient) DoConnect(ctx context.Context, inboxState *response.InboxInitialStateResponse) {
 	go tc.syncChannels(ctx, inboxState.InboxInitialState)
```

Comment:
This can probably be changed to be synchronous at the same time to fix the potential races between sync and polling